### PR TITLE
Upkeep 2024-10

### LINKED ISSUE #6155: Upkeep for ggplot2 (2024)
### To begin

* [x] `pr_init("upkeep-2024-10")`

### Pre-history

* [ ] `usethis::use_readme_rmd()`
* [x] `usethis::use_roxygen_md()`
* [x] `usethis::use_github_links()`
* [ ] `usethis::use_pkgdown_github_pages()`
* [x] `usethis::use_tidy_github_labels()`
* [ ] `usethis::use_tidy_style()`
* [x] `urlchecker::url_check()`

### 2020

* [x] `usethis::use_package_doc()`
* [x] `usethis::use_testthat(3)`
* [x] Align the names of `R/` files and `test/` files

### 2021

* [x] Remove check environments section from `cran-comments.md`
* [x] Use lifecycle instead of artisanal deprecation messages

### 2022

* [x] Handle and close any still-open `master` --> `main` issues
* [x] `usethis:::use_codecov_badge("tidyverse/ggplot2")`
* [x] Update pkgdown site using instructions at <https://tidytemplate.tidyverse.org>
* [x] Update lifecycle badges with more accessible SVGs: `usethis::use_lifecycle()`

### 2023

* [x] Update copyright holder in DESCRIPTION: `person("Posit Software, PBC", role = c("cph", "fnd"))`
* [x] Run `devtools::document()` to re-generate package-level help topic with DESCRIPTION changes
* [x] `usethis::use_tidy_logo(); pkgdown::build_favicons(overwrite = TRUE)`
* [x] `usethis::use_tidy_coc()`
* [ ] Modernize citation files; see updated `use_citation()`
* [x] Use `pak::pak("tidyverse/ggplot2")` in README
* [ ] Consider running `usethis::use_tidy_dependencies()` and/or replace compat files with `use_standalone()`
* [x] Use cli errors or [file an issue](new) if you don't have time to do it now
* [x] `usethis::use_standalone("r-lib/rlang", "types-check")` instead of home grown argument checkers;
or [file an issue](new) if you don't have time to do it now
* [x] Add alt-text to pictures, plots, etc; see https://posit.co/blog/knitr-fig-alt/ for examples

### 2024

* [x] Review pkgdown output to find missing alt-text
* [x] Switch to new quarto chunk style: https://github.com/r-lib/usethis/issues/1913
* [x] Switch to `expect_snapshot(error = TRUE)` for class-less `expect_error()`: https://github.com/r-lib/usethis/issues/1918
* [x] Record upkeep date (https://github.com/r-lib/usethis/issues/1916)

### To finish

* [x] `usethis::use_mit_license()`
* [x] `usethis::use_package("R", "Depends", "4.0")`
* [x] `usethis::use_tidy_description()`
* [x] `usethis::use_tidy_github_actions()`
* [x] `devtools::build_readme()`
* [x] [Re-publish released site](https://pkgdown.r-lib.org/dev/articles/how-to-update-released-site.html) if needed

<sup>Created on 2024-10-24 with `usethis::use_tidy_upkeep_issue()`, using [usethis v3.0.0.9000](https://usethis.r-lib.org)</sup>

### PR DESCRIPTION
**Pull request by teunbrand**
This PR aims to fix #6155.

Marked as a draft for now as things are checked off.

### CODE REVIEW

**Comment by teunbrand** [comment]
The following bullet:

>  Align the names of R/ files and test/ files

Was only exercised for `guide-legend.R`, `guide-colorbar.R`, `guide-axis.R` and `guide-.R`, mostly because I found  the 'one file for all guide tests' to be hard to grok. These tests are still skipped on CRAN.

File: LICENSE [hunk 1]
```diff
@@ -1,2 +1,2 @@
 YEAR: 2024
-COPYRIGHT HOLDER: ggplot2 core developer team
+COPYRIGHT HOLDER: ggplot2 authors
```
Comment (thomasp85) [inline LICENSE:2]: ggplot2 is special and should keep the prior wording

File: LICENSE.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # MIT License
 
-Copyright (c) 2020 ggplot2 authors
+Copyright (c) 2024 ggplot2 authors
```
Comment (thomasp85) [inline LICENSE.md:3]: this should be reworded to match the LICENSE file

**Review by thomasp85 — COMMENTED** [review]
Seems good and non-controversial. But we should change `expect_warning()` to `expect_snapshot()` while we are at it

File: LICENSE (hunk 1 above)
Comment (teunbrand) [reply LICENSE:2]: I didn't know, but then we should omit `use_mit_licence()` in future upkeep issues.

File: tests/testthat/test-aes.R [hunk 3]
```diff
@@ -48,8 +48,10 @@ test_that("aes evaluated in environment where plot created", {
   df <- data_frame(x = 1, y = 1)
   p <- ggplot(df, aes(foo, y)) + geom_point()
 
-  # Accessing an undefined variable should result in error
-  expect_error(get_layer_data(p), "'foo' not found")
+  test_that("accessing an undefined variable results in an error", {
+    skip_if(getRversion() <= "4.4.0")
+    expect_snapshot(get_layer_data(p), error = TRUE)
+  })
```
Comment (thomasp85) [inline tests/testthat/test-aes.R:54]: Is it intended you put a `test_that()` call inside another `test_that()` call?

**Review by thomasp85 — APPROVED** [review]

File: tests/testthat/test-aes.R (hunk 3 above)
Comment (teunbrand) [reply tests/testthat/test-aes.R:54]: Yes, that way I can conditionally skip a single expectation without throwing out all other expectations in a test. I improvised this to skip some snapshots on older R versions. This was due to error formatting that has changed across R versions.

**Comment by teunbrand** [comment]
Thanks for the review Thomas!

**Comment by changberg** [comment]
nice!

**Pull request merged**